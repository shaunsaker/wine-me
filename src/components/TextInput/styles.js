import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

// @WADE: error and success styles are commented out. If input validation is not needed, let me know because it will be better to remove the logic for that as opposed to just removing the styles

const styles = StyleSheet.create({
  input: {
    backgroundColor: styleConstants.colors.white,
    fontSize: styleConstants.fonts.sizes.regular,
    color: styleConstants.colors.primary,
    marginBottom: 5, // @WADE: rather let the parents control this components layout and give the parent the marginBottom
    textAlign: 'center',
    borderWidth: 1,
    borderColor: styleConstants.colors.dividerColor,
    borderRadius: 5,
    padding: 15,
    textAlignVertical: 'center',
  },
  error: {
    backgroundColor: 'rgba(255, 0, 0, 0.2)',
    color: styleConstants.colors.danger,
    borderColor: 'rgba(255, 0, 0, 0.5)',
  },
  success: {
    // color: styleConstants.colors.success,
    // backgroundColor: styleConstants.colors.success,
  },
});

export default styles;
