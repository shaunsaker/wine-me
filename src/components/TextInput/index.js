import React from 'react';
import PropTypes from 'prop-types';
import { TextInput } from 'react-native';

import styleConstants from '../../styleConstants';
import styles from './styles';

/*
  NOTE: This needs to be a functional component so that we can pass a ref prop
*/
export default class TextInputComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  static propTypes = {
    placeholder: PropTypes.string,
    value: PropTypes.string,
    handleChange: PropTypes.func,
    hasError: PropTypes.bool,
    hasSuccess: PropTypes.bool,
    keyboardType: PropTypes.string,
    handleSubmit: PropTypes.func,
    returnKeyType: PropTypes.string,
    secureTextEntry: PropTypes.bool,
    testID: PropTypes.string,
    inputRef: PropTypes.func,
    multiline: PropTypes.bool,
    style: TextInput.propTypes.style,
  };

  static defaultProps = {};

  render() {
    const {
      placeholder,
      value,
      handleChange,
      hasError,
      hasSuccess,
      keyboardType,
      handleSubmit,
      returnKeyType,
      secureTextEntry,
      testID,
      inputRef,
      multiline,
      style,
    } = this.props;

    return (
      <TextInput
        placeholder={placeholder}
        placeholderTextColor={styleConstants.colors.secondaryText}
        value={value}
        onChangeText={handleChange}
        style={[styles.input, hasError && styles.error, hasSuccess && styles.success, style]}
        keyboardType={keyboardType}
        onSubmitEditing={handleSubmit}
        returnKeyType={returnKeyType}
        secureTextEntry={secureTextEntry}
        testID={testID}
        ref={inputRef}
        multiline={multiline}
        underlineColorAndroid="transparent"
      />
    );
  }
}
