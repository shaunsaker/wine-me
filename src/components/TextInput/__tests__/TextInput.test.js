import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from '..';

describe('TextInput', () => {
  const placeholder = 'Test';
  const value = 'Foo';
  const handleChange = jest.fn();
  const keyboardType = 'email-address';
  const handleSubmit = jest.fn();
  const returnKeyType = 'done';
  const testID = 'test';
  const inputRef = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<TextInput />);

      expect(component).toMatchSnapshot();
    });

    it('renders with all props', () => {
      const component = renderer.create(
        <TextInput
          placeholder={placeholder}
          value={value}
          handleChange={handleChange}
          keyboardType={keyboardType}
          handleSubmit={handleSubmit}
          returnKeyType={returnKeyType}
          testID={testID}
          inputRef={inputRef}
          hasError
          hasSuccess
          secureTextEntry
          multiline
        />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
