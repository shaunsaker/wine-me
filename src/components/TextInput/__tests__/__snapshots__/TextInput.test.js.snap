// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders renders with all props 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="email-address"
  multiline={true}
  onChangeText={[MockFunction]}
  onSubmitEditing={[MockFunction]}
  placeholder="Test"
  returnKeyType="done"
  secureTextEntry={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E0E0E0",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#211E1C",
        "fontSize": 18,
        "marginBottom": 5,
        "padding": 15,
        "textAlign": "center",
        "textAlignVertical": "center",
      },
      Object {
        "backgroundColor": "rgba(255, 0, 0, 0.2)",
        "borderColor": "rgba(255, 0, 0, 0.5)",
        "color": "#d32f2f",
      },
      Object {},
      undefined,
    ]
  }
  testID="test"
  underlineColorAndroid="transparent"
  value="Foo"
/>
`;

exports[`TextInput renders renders with minimum required props 1`] = `
<TextInput
  allowFontScaling={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E0E0E0",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#211E1C",
        "fontSize": 18,
        "marginBottom": 5,
        "padding": 15,
        "textAlign": "center",
        "textAlignVertical": "center",
      },
      undefined,
      undefined,
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
